Owl/app/src/main/java/com/example/owl/ui/OwlApp.kt

```diff
@@ -16,21 +16,98 @@
+import androidx.navigation.NavController
+import androidx.navigation.compose.KEY_ROUTE
+import androidx.navigation.compose.currentBackStackEntryAsState
+import androidx.navigation.compose.navigate
+import androidx.navigation.compose.rememberNavController
+import com.example.owl.ui.courses.CourseTabs
+import com.example.owl.ui.theme.BlueTheme
 import com.example.owl.ui.utils.ProvideImageLoader
 import com.google.accompanist.insets.ProvideWindowInsets
+import com.google.accompanist.insets.navigationBarsHeight
+import com.google.accompanist.insets.navigationBarsPadding
 
 @Composable
-fun OwlApp(backDispatcher: OnBackPressedDispatcher) {
+fun OwlApp(finishActivity: () -> Unit) {
+    ProvideWindowInsets {
+        ProvideImageLoader {
+            BlueTheme {
+                val tabs = CourseTabs.values()
+                val navController = rememberNavController()
+                OwlScaffold(
+                    bottomBar = { OwlBottomBar(navController = navController, tabs) }
+                ) { innerPaddingModifier ->
+                    NavGraph(
+                        finishActivity = finishActivity,
+                        navController = navController,
+                        modifier = innerPaddingModifier
+                    )
+                }
+            }
+        }
+    }
+}
+
+@Composable
+fun OwlBottomBar(navController: NavController, tabs: Array<CourseTabs>) {
 
-    CompositionLocalProvider(LocalBackDispatcher provides backDispatcher) {
-        ProvideWindowInsets {
-            ProvideImageLoader {
-                NavGraph()
+    val navBackStackEntry by navController.currentBackStackEntryAsState()
+    val currentRoute = navBackStackEntry?.arguments?.getString(KEY_ROUTE)
+        ?: CourseTabs.FEATURED.route
+
+    if (currentRoute in CourseTabs.values().map { it.route }) {
+        BottomNavigation(
+            Modifier.navigationBarsHeight(additional = 56.dp)
+        ) {
+            tabs.forEach { tab ->
+                BottomNavigationItem(
+                    icon = { Icon(painterResource(tab.icon), contentDescription = null) },
+                    label = { Text(stringResource(tab.title).toUpperCase()) },
+                    selected = currentRoute == tab.route,
+                    onClick = {
+                        if (tab.route != currentRoute) {
+                            navController.navigate(tab.route) {
+                                popUpTo = navController.graph.startDestination
+                                launchSingleTop = true
+                            }
+                        }
+                    },
+                    alwaysShowLabel = false,
+                    selectedContentColor = MaterialTheme.colors.secondary,
+                    unselectedContentColor = LocalContentColor.current,
+                    modifier = Modifier.navigationBarsPadding()
+                )
             }
         }
     }
 }
+
+@Composable
+fun OwlScaffold(
```

Comment:
Just use `Scaffold`? The only thing this is really doing is wrapping the content padding into a modifier.